src/CodeGen_PTX_Dev.cpp

```diff
@@ -791,6 +813,334 @@ bool CodeGen_PTX_Dev::supports_atomic_add(const Type &t) const {
+                return MatrixMultiplyInfo{};
+            }
+
+            // If the swapped pattern works, swap load_a and load_b as matrix multiplies are non-commutative
+            std::swap(load_a, load_b);
+        }
+
+        // Check if the k_var_name is present in the expressions for load_a and load_b
+        const Variable *load_a_var_1 = matches_a[0].as<Variable>();
+        const Variable *load_a_var_2 = matches_a[1].as<Variable>();
+        const Variable *load_b_var_1 = matches_b[0].as<Variable>();
+        const Variable *load_b_var_2 = matches_b[1].as<Variable>();
+
+        const std::string &k_var_name = loop->name;
+
+        // TODO: Can this checks be improved?
+        const bool load_a_ok = (load_a_var_1 && load_a_var_1->name == k_var_name) || (load_a_var_2 && load_a_var_2->name == k_var_name);
+        const bool load_b_ok = (load_b_var_1 && load_b_var_1->name == k_var_name) || (load_b_var_2 && load_b_var_2->name == k_var_name);
+
+        if (!load_a_ok || !load_b_ok) {
+            return MatrixMultiplyInfo{};
+        }
+
+        // This matches A(k, y) * B(x, k) where both A and B
+        MatrixMultiplyInfo matMulInfo;
+        matMulInfo.A = load_a;
+        matMulInfo.B = load_b;
+        matMulInfo.C = load_c;
+        return matMulInfo;
+    }
+
+    return MatrixMultiplyInfo{};
+}
+
+ExtractTensorCoreOperations::ExtractTensorCoreOperations() {
+    thread_id_y = Call::make(Int(32), CodeGen_PTX_Dev::simt_intrinsic(".__thread_id_y"), std::vector<Expr>(), Call::Extern);
+    thread_id_x = Call::make(Int(32), CodeGen_PTX_Dev::simt_intrinsic(".__thread_id_x"), std::vector<Expr>(), Call::Extern);
+    block_id_y = Call::make(Int(32), CodeGen_PTX_Dev::simt_intrinsic(".__block_id_y"), std::vector<Expr>(), Call::Extern);
+    block_id_x = Call::make(Int(32), CodeGen_PTX_Dev::simt_intrinsic(".__block_id_x"), std::vector<Expr>(), Call::Extern);
+    block_dim_y = Call::make(Int(32), CodeGen_PTX_Dev::simt_intrinsic(".__block_dim_y"), std::vector<Expr>(), Call::Extern);
+    block_dim_x = Call::make(Int(32), CodeGen_PTX_Dev::simt_intrinsic(".__block_dim_x"), std::vector<Expr>(), Call::Extern);
+    block_size = block_dim_x * block_dim_y;
+}
+
+Stmt ExtractTensorCoreOperations::visit(const For *loop) {
+    const bool is_gpu_thread_var = CodeGen_GPU_Dev::is_gpu_thread_var(loop->name);
+    if (is_const_zero(loop->min) && is_const(loop->extent)) {
+        const int32_t loop_extent_value = loop->extent.as<IntImm>()->value;
+
+        if (is_gpu_thread_var) {
+            if (ends_with(loop->name, ".__thread_id_y")) {
+                wmma_M = loop_extent_value;
+            } else if (ends_with(loop->name, ".__thread_id_x")) {
+                wmma_N = loop_extent_value;
+            }
+        } else {
+            wmma_K = loop_extent_value;
+
+            // Shape m16n16k16
+            // TODO: Note this won't work unless the extent of the inner loop is constant
```

Comment:
I think this is reasonable but it will require more changes in the way that the wmma ops are extracted, i.e. splitting K will generate the wrong kernel with this PR. This is of course something I want to address before merging so I will come back to this soon.